internal/server/acme/acme.go

```diff
@@ -99,96 +101,147 @@ func UpdateCertificate(s *state.State, provider ChallengeProvider, clustered boo
 		return nil, nil
 	}
 
-	// Generate new private key for user. This key needs to be different from the server's private key.
-	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	if err != nil {
-		return nil, fmt.Errorf("Failed generating private key for user account: %w", err)
-	}
+	if challengeType == "DNS-01" {
+		provider, environment, resolvers := s.GlobalConfig.ACMEDNS()
 
-	user := user{
-		Email: email,
-		Key:   privateKey,
-	}
+		if provider == "" {
+			return nil, fmt.Errorf("DNS-01 challenge type requires acme.dns.provider configuration key to be set")
+		}
 
-	config := lego.NewConfig(&user)
+		tmpDir, err := os.MkdirTemp("", "lego")
+		if err != nil {
+			return nil, fmt.Errorf("Failed to create temporary directory: %w", err)
+		}
+		defer os.RemoveAll(tmpDir)
+
+		args := []string{
+			"--accept-tos",
+			"--dns", provider,
+			"--domains", domain,
+			"--email", email,
+			"--path", tmpDir,
+			"--server", caURL,
+		}
 
-	if caURL != "" {
-		config.CADirURL = caURL
-	} else {
-		// Default URL for Let's Encrypt
-		config.CADirURL = "https://acme-v02.api.letsencrypt.org/directory"
-	}
+		if len(resolvers) > 0 {
+			for _, resolver := range resolvers {
+				args = append(args, "--dns.resolvers", resolver)
+			}
+		}
 
-	config.Certificate.KeyType = certcrypto.RSA2048
+		args = append(args, "run")
 
-	client, err := lego.NewClient(config)
-	if err != nil {
-		return nil, fmt.Errorf("Failed to create new client: %w", err)
-	}
+		cmd := exec.Command("lego", args...)
+		cmd.Env = append(os.Environ(), environment...)
 
-	err = provider.RegisterWithSolver(client.Challenge)
-	if err != nil {
-		return nil, fmt.Errorf("Failed setting challenge provider: %w", err)
-	}
+		logger.Info("Running lego command", logger.Ctx{"cmd": cmd.String()})
+		err = cmd.Run()
```

Comment:
Should be able to use our `subprocess.RunCommandSplit` for this instead.

`_, _, err := subprocess.RunCommandSplit(ctx, append(os.Environ(), environment), nil, "lego", args...)`